addons/website_sale/static/src/js/website_sale.js

```diff
@@ -655,4 +657,41 @@ sAnimations.registry.websiteSaleCart = sAnimations.Class.extend({
         $(ev.currentTarget).closest('tr').find('.js_quantity').val(0).trigger('change');
     },
 });
+
+sAnimations.registry.productsSearchBar = sAnimations.Class.extend({
+    selector: '.o_wsale_products_search_bar_input',
+    xmlDependencies: ['/website_sale/static/src/xml/website_sale_utils.xml'],
+    read_events: {
+        'click .oe_search_button': '_onSubmitProductsSearch',
+    },
+
+    /**
+     * @override
+     */
+    start: function () {
+        this.$searchProductsInput = this.$target.find('.o_wsale_search_products');
+        wUtils.autocompleteWithProducts(this, this.$searchProductsInput, {
+            displayDescription: this.$searchProductsInput.attr('data-display-description') === 'true',
+            displayPrice: this.$searchProductsInput.attr('data-display-price') === 'true',
+            displayImage: this.$searchProductsInput.attr('data-display-image') === 'true',
+            orderBy: this.$searchProductsInput.attr('data-order-by'),
+        });
+        return this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {Event} ev
+     */
+    _onSubmitProductsSearch: function (ev) {
+        if (window.location.pathname !== '/shop') {
```

Comment:
How is the shop working currently ? Should not it be using this code too ? (or maybe you should simply have moved the current logic here ? I see that the shop is using a form, I think your snippet should do the same, then there would not be any JS associated.